packages/block-editor/src/utils/object.js

```diff
@@ -112,6 +113,8 @@ export function setImmutably( object, path, value ) {
 	return newObject;
 }
 
+const stringToPath = memoize( ( path ) => path.split( '.' ) );
```

Comment:
This has the drawback of taking up memory that is never released. Maybe harmless here because it's small strings that probably get reused, but still worth nothing. I guess this is a weird case where a max size doesn't make sense. See also https://github.com/WordPress/gutenberg/pull/53406/files#r1286726012

Reply:
Removing in #56711. I remember when profiling, the memoization didn't have any impact on performance anyway.